content/ui-ux.md

```diff
@@ -0,0 +1,514 @@
+---
+title: User Interfaces and User Experience
+---
+
+After reading this guide, you'll know:
+
+1. How to build re-usable client side components in any templating language
+2. How to build a styleguide to allow you to visually test such pure components
+3. Patterns for building front end components in a performant way in Meteor
+4. How to design responsively across device sizes, accessibly for different users, and universally across languages.
+5. How to build components that can cope with a variety of different data sources
+6. How to use animation to keep users informed of changes
+
+## Reusable components
+
+Regardless of the rendering library that you are using, there are some patterns in how you build your User Interface (UI) that will help make your app's code easier to understand, test, and maintain. These patterns, much like general patterns of modularity, revolve around making the interfaces of your UI elements very clear, and avoiding using techniques that bypass these known interfaces.
+
+In this article, we'll refer to the elements in your user interface as "components". Although in some systems, you may refer to them as "templates", it can be a good idea to think of them as something more modular like a component which has an API, rather than a template which is usually seen in a looser way.
+
+To begin with, let's consider two categories of components that are useful to think about:
+
+### Pure Components
+
+A "pure" component is a component which doesn't rely on anything apart from it's inputs (it's *template arguments* in Blaze, or *props* in React) to render. 
+
+In Meteor, specifically this means a component which does not access data from any global sources (typically either Collections or Stores). For instance, in the Todos example app, the `todosItem` template takes in the todo that it is rendering and does not ever look directly in the `Todos` collection.
+
+The advantages of pure components are the following:
+
+ 1. They are easy to reason about---you don't need to understand how the data in the global store changes, simply how the arguments to the component change.
+
+ 2. They are easy to test---you don't need to be careful about the environment you render them in, all you need to do is provide the right arguments.
+
+ 3. They are easy to styleguide---as we'll see in the next section, when styleguiding components, a clean environment makes things much easier to work with.
+
+ 4. They are a lot more flexible, and thus can be re-used without requiring re-working.
```

Comment:
I think this can be phrased as, you know exactly what they do and what dependencies you need to provide for them to work in different environments. "Flexible" is quite vague, as is "re-working".